core/src/main/resources/jenkins/model/Jenkins/_safeRestart.jelly

```diff
@@ -32,14 +32,26 @@ THE SOFTWARE.
 			<j:choose>
 				<j:when test="${app.lifecycle.canRestart()}">
 					<form method="post" action="safeRestart">
+						<f:entry title="${%Restart Message}">
+						  <f:textbox name="message" value="${%The Jenkins Controller is restarting gently. Running jobs can continue. No new jobs are accepted.}"/>
+						</f:entry>
 						${%confirmRestart}
-						<f:submit value="${%Yes}" />
+						<f:bottomButtonBar>
+							<f:submit clazz="jenkins-!-destructive-color" value="${%Restart}" />
```

Comment:
I oriented myself at "prepareShutdown", since that also pauses jobs.
https://github.com/jenkinsci/jenkins/blob/5870838f690eaa06762630eceb21c40d3df0ceb8/core/src/main/resources/jenkins/management/ShutdownLink/index.groovy